### ISSUE TITLE
Optimize field line following with jax.lax.scan

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#61]
## Background
PR #60 implements field line following for computing true k∥ spectra. The current implementation uses a Python for loop, which is not JIT-compiled and limits performance for large-scale production runs.

## Current Implementation
```python
for step in range(n_steps_max):
    if pos[2] >= grid.Lz / 2:
        break
    # RK2 integration with interpolation...
    trajectory_list.append(pos.copy())
```

## Performance Impact
- Cannot be JIT-compiled due to Python loop
- Dynamic trajectory list accumulation
- Each field line takes ~7 ms on 256³ grid
- For ensemble statistics (100+ field lines), this adds up

## Proposed Solution
Rewrite using `jax.lax.scan` for JIT-compiled fixed-step integration:

```python
def scan_body(carry, _):
    pos = carry
    # RK2 step (same logic as current)
    pos_new = rk2_step(pos, B_fine, dz, grid)
    return pos_new, pos_new

# Fixed number of steps
n_steps = int(jnp.ceil(grid.Lz / dz))
pos0 = jnp.array([x0, y0, -grid.Lz/2])
_, trajectory = jax.lax.scan(scan_body, pos0, None, length=n_steps)
```

## Benefits
- Full JIT compilation → 10-100× speedup
- Memory-efficient (pre-allocated trajectory array)
- Cleaner code (functional style)

## Considerations
- Requires fixed number of steps (no early termination)
- May need to filter trajectory points outside domain
- Interpolation functions must be compatible with scan

## References
- JAX lax.scan documentation
- Similar patterns in timestepping.py (RK4 uses scan)
- Issue #25: Field line following diagnostics

## Acceptance Criteria
- [ ] Rewrite `follow_field_line` using `jax.lax.scan`
- [ ] Maintain API compatibility (same inputs/outputs)
- [ ] All existing tests pass
- [ ] Benchmark shows significant speedup
- [ ] Add performance comparison in docstring

**Comment by anjor** [comment]
Closing as duplicate of Issue #62, which was resolved with the implementation of spectral interpolation via FFT padding (2× resolution) for field line following. See `src/krmhd/diagnostics.py` for the complete implementation.
